braid/residual.c

```diff
@@ -79,24 +106,53 @@ _braid_FASResidual(braid_Core        core,
    _braid_Grid      **grids  = _braid_CoreElt(core, grids);
    braid_Int          ilower = _braid_GridElt(grids[level], ilower);
    braid_BaseVector  *fa     = _braid_GridElt(grids[level], fa);
+   braid_BaseVector  *va     = _braid_GridElt(grids[level], va);
+   // braid_Basis       *ba     = _braid_GridElt(grids[level], ba);
 
-   braid_Int        ii;
 
-   _braid_Residual(core, level, index, ustop, r);
-   if (level == 0)
+   braid_Int ii = index-ilower;
+
+   /* && short circuits, so (fa[ii] != NULL) is never evaluated if (level == 0) */
+   braid_Int delta_correct = _braid_CoreElt(core, delta_correct)
+                             && (_braid_CoreElt(core, niter) >= _braid_CoreElt(core, delta_defer_iter))
+                             && (level > _braid_CoreElt(core, delta_defer_lvl))
```

Comment:
Should this be 

level > delta_defer_lvl

or

level >= delta_defer_lv

? 
I think that an above statement uses ">"